regressiontests/regression_test.go

```diff
@@ -0,0 +1,391 @@
+//go:build regression
+// +build regression
+
+package regressiontests
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"flag"
+	"fmt"
+	"io"
+	"net/http"
+	"os"
+	"os/exec"
+	"testing"
+	"time"
+
+	"github.com/bitcoin-sv/block-headers-service/config"
+	"github.com/bitcoin-sv/block-headers-service/domains"
+	"github.com/bitcoin-sv/block-headers-service/internal/chaincfg"
+	"github.com/bitcoin-sv/block-headers-service/transports/http/endpoints/api/headers"
+	"github.com/bitcoin-sv/block-headers-service/transports/http/endpoints/api/merkleroots"
+	"github.com/bitcoin-sv/block-headers-service/transports/http/endpoints/api/tips"
+	"github.com/testcontainers/testcontainers-go"
+	"github.com/testcontainers/testcontainers-go/wait"
+)
+
+const localHTTPServerURL = "http://localhost:8080"
+const whatsonchainAPIURL = "https://api.whatsonchain.com/v1/bsv/main/chain/tips"
+
+type WhatsOnChainForkTip struct {
+	Height    int    `json:"height"`
+	Hash      string `json:"hash"`
+	BranchLen int    `json:"branchlen"`
+	Status    string `json:"status"`
+}
+
+var dbEngine string
+
+func init() {
+	flag.StringVar(&dbEngine, "dbEngine", "sqlite", "The database engine to use in tests (postgres or sqlite)")
+}
+
+func TestApplicationIntegration(t *testing.T) {
```

Comment:
I suppose that this test will be detected as normal unit test. In would make standard unit-tests check much longer. 

Maybe we should use sth like that:
```golang
if testing.Short() {
	t.Skip("skipping regression test")
}
```

Reply:
In my opinion it shouldn't be detected as normal test and run along with other tests - I followed the same approach as in spv-wallet and added:
```
//go:build regression
// +build regression
```
so we can access those tests using this tag.

Reply:
ahh ok 👍 
Nice 💯